homeassistant/components/zone.py

```diff
@@ -44,6 +44,7 @@ def active_zone(hass, latitude, longitude, radius=0):
         within_zone = zone_dist - radius < zone.attributes[ATTR_RADIUS]
         closer_zone = closest is None or zone_dist < min_dist
         smaller_zone = (zone_dist == min_dist and
+                        zone.attributes[ATTR_RADIUS] != 0 and
```

Comment:
GPS isn't precise for small regions - so you can't distinguish between (say) rooms.

But with ibeacons you can be precise about where you are. However you then you don't want gps picking the region (because it'll get it wrong).

So radius 0 is an attempt to say 'don't pick this region via gps'. But it didn't work without the check I added. There may be better fixes.

It would also be nice to be able to have room zones marked as 'consider home', so you can nest iBeacon zones for rooms inside a larger home zone.